editoast/src/views/v2/timetable/stdcm.rs

```diff
@@ -233,6 +246,87 @@ async fn stdcm(
     Ok(Json(stdcm_response))
 }
 
+/// get the maximum run time, compute it if unspecified.
+/// returns an enum with either the result or a SimulationResponse if it failed
+async fn get_max_run_time(
+    db_pool: Arc<DbConnectionPool>,
+    redis_client: Arc<RedisClient>,
+    core_client: Arc<CoreClient>,
+    data: &STDCMRequestPayload,
+    infra: &Infra,
+    rolling_stock: &RollingStockModel,
+) -> Result<MaxRunningTimeResult> {
+    if data.maximum_run_time.is_some() {
+        return Ok(MaxRunningTimeResult::MaxRunningTime {
+            value: data.maximum_run_time.unwrap(),
+        });
+    }
```

Comment:
The `if` supports conditional destructuring in Rust so you should be able to do the following instead.
```suggestion
    if let Some(maximum_run_time) = data.maximum_run_time {
        return Ok(MaxRunningTimeResult::MaxRunningTime {
            value: maximum_run_time,
        });
    }
```